### PR TITLE
Update Documentation and Build Settings

### LINKED ISSUE #987: CLI Distribution Design Discussion
### Summary
This task is a place for @wdanilo to gather his feedback regarding the Enso CLI distribution schemes.

### Value
It's important for @wdanilo to be on board with the proposed design, therefore we need a single place where feedback can be gathered.

### Specification
- [x] Read all the documents contained in: https://dev.enso.org/docs/enso/distribution/ (i.e. the 5 subsections there).
- [x] Voice any concerns as comments on this issue.
- [x] Let's have a further discussion.
- [x] Incorporate the results of the discussion into the design documentation.

### Acceptance Criteria & Test Cases
A design consensus is reached and has been incorporated into the documentation.

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

After the CLI distribution design discussion #987, some documentation had to be updated to reflect the decisions made.

Closes #987.

### Important Notes

- It renames the `enso.jar` to `runner.jar`, but the simple launcher script that is put in the `bin` directory is still `enso`.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: docs/distribution/distribution.md [hunk 1]
```diff
@@ -12,52 +12,130 @@ be structured and how it should behave.
...
   - [Resolvers](#resolvers)
 
 <!-- /MarkdownTOC -->
 
-## Enso Home Layout
-All of Enso's binaries, packages, etc. are installed into a directory in
-the user's home directory. For macOS and linux distributions that's `~/.enso`,
-by default. The distribution is fully portable, so it never makes any
-assumptions about actually being placed in any particular directory.
+## Universal Launcher Script
```
Comment (iamrecursion) [inline docs/distribution/distribution.md]: I'm not sure we need to call this a 'script' any more. The "Universal Launcher" is fine!

File: docs/distribution/distribution.md [hunk 2]
```diff
@@ -12,52 +12,130 @@ be structured and how it should behave.
...
+The [universal launcher script](./launcher.md) should be able to launch the
+proper version of Enso executable based on the version specified in the project
+being run, or use the default version if none specified. It should also be able
+to launch other Enso components, provided as
+[plugins](./launcher.md#running-plugins).
+
+**Note**
+This launcher is under development. Until it is in a ready-to-use state, the
+Enso version packages provide simple launcher scripts in the `bin` directory of
+that package. They will be removed when the universal launcher matures.
```
Comment (iamrecursion) [inline docs/distribution/distribution.md]: This should be a `>` block. Why would they be removed entirely? They still could help the launcher start the enso artefacts, no?

File: docs/distribution/distribution.md [hunk 3]
```diff
@@ -12,52 +12,130 @@ be structured and how it should behave.
...
+If not set, each of these three environment variables defaults to the following
+value, depending on the system:
+
+|                         | Linux                                                              | macOS                                         | Windows                      |
+|-------------------------|--------------------------------------------------------------------|-----------------------------------------------|------------------------------|
+| `ENSO_DATA_DIRECTORY`   | `$XDG_DATA_HOME/enso/` which defaults to `$HOME/.local/share/enso` | `$HOME/Library/Application Support/org.enso/` | `%LocalAppData%/enso`        |
+| `ENSO_CONFIG_DIRECTORY` | `$XDG_CONFIG_HOME/enso/` which defaults to `$HOME/.config/enso`    | `$HOME/Library/Preferences/org.enso/`         | `%LocalAppData%/enso/config` |
+| `ENSO_BIN_DIRECTORY`    | `$XDG_BIN_HOME` which defaults to `$HOME/.local/bin`               | `$HOME/.local/bin`                            | `%LocalAppData%/enso/bin`    |
+
+### Converting a Portable Distribution into Installed
```
Comment (iamrecursion) [inline docs/distribution/distribution.md]:
```suggestion
### Installing from a Portable Distribution
```

**Review by iamrecursion — APPROVED** [review]
Super, super happy with this. Some minor cosmetic fixes

**Review by kustosz — APPROVED** [review]
I love that we'll get the portable distribution as well <3

File: docs/distribution/distribution.md (hunk 2 above)
Comment (radeusgd) [reply docs/distribution/distribution.md]: @kustosz told me that they should be removed after the universal launcher script matures, as they were more of a hack.

The launcher should most likely handle launching itself, as it must have fine-grained control on the used JVM and for example, additional JVM parameters.

**Review by 4e6 — APPROVED** [review]

**Review by wdanilo — APPROVED** [review]
I love it <3

**Pull request merged**